lib/administrator.rb

```diff
@@ -0,0 +1,214 @@
+
+require 'time'
+require 'pry'
+require 'awesome_print'
+
+require_relative 'room'
+require_relative 'reservation'
+require_relative 'block'
+
+module Hotel
+  class Administrator
+
+    attr_reader :rooms, :reservations, :blocks
+
+    def initialize
+      @rooms = room_list
+      @reservations = []
+      @blocks = []
+    end
+
+    def room_list
+      room_ids = (1..20).to_a
+      room_list = []
+      room_ids.each do |room_id|
+        room_list << Room.new(room_id)
+      end
+      return room_list
+    end
+
+    def find_room(room_id)
+      check_room_id(room_id)
+      rooms.find{ |room| room.room_id == room_id}
+    end
+
+    def reservations_on_date(date)
+      check_date(date)
+
+      reservations.select { |reservation| (reservation.check_out_date == date) ||  (reservation.block_of_dates.include?(date) == true) }
```

Comment:
Question:  If the room is being checked out of on a given date, can other people check in on that date, usually yes.